From 286ad4688c2a512b8f5b3739763ae7ba7b1821c4 Mon Sep 17 00:00:00 2001
From: Andy Ross <andrew.j.ross@intel.com>
Date: Tue, 18 Nov 2014 11:50:01 -0800
Subject: [PATCH] ov5693: Better initial focus position

Some code (like CTSVerifier) uses the camera in manual focus mode but
never moves the focus and relies on the defaults being sane.  Our
driver was slamming the VCM all the way to zero (near-field/macro
focus), which is suboptimial.

Note that this in-driver VCM code will soon be replaced with
in-progress work on ACPI-probed VCM drivers which will need the same
treatment.

Tracked-On: https://jira01.devtools.intel.com/browse/GMINL-3862
Change-Id: I005e3035109b93cbec8116f293b16a0db446ed72
Signed-off-by: Andy Ross <andrew.j.ross@intel.com>
---
 drivers/external_drivers/camera/drivers/media/i2c/ov5693.c | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/drivers/external_drivers/camera/drivers/media/i2c/ov5693.c b/drivers/external_drivers/camera/drivers/media/i2c/ov5693.c
index eeb961a..5f4ed2d 100755
--- a/drivers/external_drivers/camera/drivers/media/i2c/ov5693.c
+++ b/drivers/external_drivers/camera/drivers/media/i2c/ov5693.c
@@ -40,6 +40,9 @@
 
 #include "ov5693.h"
 
+/* Focus for the AD5823 should start in the middle of the field */
+#define INIT_FOCUS_POS 350
+
 static int vcm_i2c_wr8(struct i2c_client *client, u8 reg, u8 val)
 {
 	int err;
@@ -1107,7 +1110,7 @@ static int ov5693_init(struct v4l2_subdev *sd)
 	N_RES = N_RES_PREVIEW;
 
 	dev->focus = 0;
-	ov5693_t_focus_abs(sd, 0);
+	ov5693_t_focus_abs(sd, INIT_FOCUS_POS);
 
 	mutex_unlock(&dev->input_lock);
 
-- 
1.9.3

